### PR TITLE
Survey creation

### PR DESCRIPTION
**Pull request by C1erlc**
Added survey creation page for only logged in users to create surveys. Users are able to add any number of multiple choice, number range, and open ended questions, as well as pick any minimum and maximum for number ranges and add choices to multiple choice questions. Users are also able to name the survey and give the text for each question, as well as for each choice in multiple choice questions.

### CODE REVIEW

**Review by Sgt-Roukoz — CHANGES_REQUESTED** [review]
Few changes you might want to make. First merge main with the branch properly, as it is not merged. And make it so that the created survey is added to the logged in user, which you can do by adding `HttpSession session` to the controller parameters, and `UserAccount user = userAccountRepository.findByUsername(session.getAttribute("username").toString());` to get the user. other than that, it should be good!

File: src/main/java/group/thirtyone/controllers/CreateSurveyController.java [hunk 1]
```diff
@@ -1,15 +1,78 @@
...
+
     @GetMapping("/create")
-    public String createSurvey(Model model, HttpSession session) {
+    public String createSurvey(Model model) {
         return "create";
     }
+
+    @PostMapping("/createSurvey")
+    @CrossOrigin
+    public String createSurveySubmit(@RequestBody String surveyJSON) {
```
Comment (Sgt-Roukoz) [inline src/main/java/group/thirtyone/controllers/CreateSurveyController.java]: Add `HttpSession session`

File: src/main/java/group/thirtyone/controllers/CreateSurveyController.java [hunk 2]
```diff
@@ -1,15 +1,78 @@
...
+        Survey survey = new Survey();
```
Comment (Sgt-Roukoz) [inline src/main/java/group/thirtyone/controllers/CreateSurveyController.java:31]: add `UserAccount user = userAccountRepository.findByUsername(session.getAttribute("username").toString());`

File: src/main/java/group/thirtyone/surveycomponents/Survey.java
Comment (Sgt-Roukoz) [inline src/main/java/group/thirtyone/surveycomponents/Survey.java:1]: Remove duplicate methods

File: src/main/resources/templates/error.html
Comment (Sgt-Roukoz) [inline src/main/resources/templates/error.html:1]: why did you remove error.html?

**Review by Sgt-Roukoz — CHANGES_REQUESTED** [review]
Make sure to properly merge main into this branch, then add the requested changes. Good job so far though

File: src/main/resources/templates/create.html [hunk 3]
```diff
@@ -1,10 +1,128 @@
...
+                console.log(formData);
+                $.ajax({
+                    type: 'POST',
+                    url: "createSurvey",
+                    dataType: 'json',
+                    contentType: "application/json; charset=utf-8",
+                    data: formData
+                });
+                var baseURL = window.location.origin;
+                window.location.href = baseURL;
```
Comment (Sgt-Roukoz) [inline src/main/resources/templates/create.html]: Change this part to this:
`var baseURL = window.location.origin;

            $('.submit').on("click", function() {
                var formData = JSON.stringify($(".createSurvey").serializeArray());
                console.log(formData);
                $.ajax({
                    type: 'POST',
                    url: baseURL+"/createSurvey",
                    dataType: 'json',
                    contentType: "application/json; charset=utf-8",
                    data: formData
                });

                setTimeout(function(){
                    window.location.href = baseURL;
                }, 15);`

Sometimes the submission ajax request does not go through because the redirection is too fast

**Comment by C1erlc** [comment]
Created survey creation page, fixed all known bugs, implemented all requested changes

**Review by Sgt-Roukoz — APPROVED** [review]

**Comment by Sgt-Roukoz** [comment]
Changes look good, survey created saves properly. Good job.
